packages/application-shell/src/components/navbar/navbar-new/navbar.tsx

```diff
@@ -103,57 +104,61 @@ const getIsSubmenuRouteActive = (
 export const ApplicationMenu = (props: ApplicationMenuProps) => {
   const [submenuVerticalPosition, setSubmenuVerticalPosition] = useState(0);
   const [isSubmenuAboveMenuItem, setIsSubmenuAboveMenuItem] = useState(false);
-  const menuItemRef = useRef<HTMLDivElement>(null);
   const submenuRef = useRef<HTMLUListElement>(null);
 
   const hasSubmenu =
     Array.isArray(props.menu.submenu) && props.menu.submenu.length > 0;
 
-  const menuItemBoundingClientRect =
-    menuItemRef.current?.getBoundingClientRect();
-  const menuItemTop = menuItemBoundingClientRect?.top || 0;
-  const menuItemBottom = menuItemBoundingClientRect?.bottom || 0;
+  const menuItemIdentifier = snakeCase(props.menu.key);
+
+  const callbackFn: IntersectionObserverCallback = useCallback((entries) => {
+    const menuItemBoundingClientRect = document
+      .querySelector(`[data-menuitem="${menuItemIdentifier}"]`)
+      ?.getBoundingClientRect();
+    const menuItemTop = menuItemBoundingClientRect?.top || 0;
+    const menuItemBottom = menuItemBoundingClientRect?.bottom || 0;
```

Comment:
Instead of using a ref now we're using `querySelector` to find the relevant menu item element. This ensures the ref used in callback is not stale when the `IntersectionObserver` instance is cached.
Many thanks to @CarlosCortizasCT for the idea 🙏